src/Speckle.Sdk/Transports/ServerUtils/ParallelServerAPI.cs

```diff
@@ -245,31 +240,44 @@ protected override void ThreadMain()
     switch (operation)
     {
       case ServerApiOperation.DownloadSingleObject:
-        var (dsoStreamId, dsoObjectId) = ((string, string))inputValue;
-        return await serialApi.DownloadSingleObject(dsoStreamId, dsoObjectId).ConfigureAwait(false);
+        var (dsoStreamId, dsoObjectId, progress) = ((string, string, Action<ProgressArgs>?))inputValue;
```

Comment:
I feel like we could clean all these tuples up with a generic class of some sort, as both the streamId and progress are consistent, we'd only need to specify if the result holds a `string`, `List<string>`, etc...

Not a blocker for merging this, it's more on the house-keeping side.